### PR TITLE
[Issue 421]Formatted Scala code and added code style template

### PR DESCRIPTION
**Pull request by ravipesala**
Add Scala code style template to the same dev/java-code-format-template.xml.
Formatted scala code along with following operations.

- Removed unnecessary semicolon
- Handled Java accessor method called as empty-paren
- Named boolean parameters
- Removed Never used variables
- Removed redundant collection conversion
- Updated Unnecassary var converted to val
- Used match case wherever it is required instead of if else
- for loop style is corrected by using indices instead of for(i <- 0 until result.length) to for(i <- result.indices)
- redundant Return keyword is removed
- Used Seq.head instead of using Seq(0) in all places
- Simplified boolean expression like if(exp == true) to if(exp)
- Used isEmpty or nonEmpty instead of using size>0 or size==0 or collections

### CODE REVIEW

**Comment by jackylk** [comment]
Thanks for working on this. Just one suggestion, better not to use for loop when it is convenient. It seems scala for loop has some [performance issue](http://stackoverflow.com/questions/16785826/why-are-scala-for-loop-comprehensions-so-very-slow-compared-to-for-loops) if not used carefully

**Comment by ravipesala** [comment]
@jackylk Thank you for reviewing it. Actually many places for loop is not required, supposed to use map. So I have replaced with map wherever possible.
 The link you provided is a good article but it mentions about  “for loop comprehensions” like reduce so very slow compared to FOR loops, it does not say scala for loop is slower.

File: integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonDataRDDFactory.scala [hunk 1]
```diff
@@ -285,25 +283,25 @@ object CarbonDataRDDFactory extends Logging {
       }
 
       var currentLoadCount = -1
-      if (carbonLoadModel.getLoadMetadataDetails().size() > 0) {
-        for (eachLoadMetaData <- carbonLoadModel.getLoadMetadataDetails().asScala) {
-          val loadCount = Integer.parseInt(eachLoadMetaData.getLoadName())
+      if (carbonLoadModel.getLoadMetadataDetails.size() > 0) {
```
Comment (jackylk) [inline integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonDataRDDFactory.scala]: change to nonEmpty

File: integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonDataRDDFactory.scala [hunk 2]
```diff
@@ -285,25 +283,25 @@ object CarbonDataRDDFactory extends Logging {
...
+        for (eachLoadMetaData <- carbonLoadModel.getLoadMetadataDetails.asScala) {
```
Comment (jackylk) [inline integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonDataRDDFactory.scala]: change to map

File: integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonDropCubeRDD.scala [hunk 3]
```diff
@@ -52,8 +49,6 @@ class CarbonDropCubeRDD[K, V](
 
       val partitionCount = partitioner.partitionCount
       for (a <- 0 until partitionCount) {
```
Comment (jackylk) [inline integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonDropCubeRDD.scala]: this is empty block?

File: integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonGlobalDictionaryRDD.scala [hunk 4]
```diff
@@ -69,17 +71,18 @@ case class PrimitiveParser(dimension: CarbonDimension,
...
   def addChild(child: GenericParser): Unit = {
     children = child
   }
+
   def parseString(input: String): Unit = {
     if (StringUtils.isNotEmpty(input)) {
       val splits = format.getSplits(input)
       if (ArrayUtils.isNotEmpty(splits)) {
-        for (i <- 0 until splits.length) {
+        for (i <- splits.indices) {
```
Comment (jackylk) [inline integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonGlobalDictionaryRDD.scala]: change to zipWithIndex

File: integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonQueryRDD.scala [hunk 5]
```diff
@@ -122,11 +118,10 @@ class CarbonSparkPartition(rddId: Int, val idx: Int,
         logInfo("Time taken to identify Blocks to scan : " + (System
           .currentTimeMillis() - startTime)
         )
-        for (j <- 0 to result.size()-1)
-        {
+        for (j <- 0 until result.size()) {
```
Comment (jackylk) [inline integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonQueryRDD.scala]: zipWithIndex

File: integration/spark/src/main/scala/org/carbondata/spark/util/GlobalDictionaryUtil.scala [hunk 6]
```diff
@@ -59,18 +59,19 @@ object GlobalDictionaryUtil extends Logging {
...
+      headers: Array[String],
+      columns: Array[String]): (Array[CarbonDimension], Array[String]) = {
     val dimensionBuffer = new ArrayBuffer[CarbonDimension]
     val columnNameBuffer = new ArrayBuffer[String]
     val dimensionsWithDict = dimensions.filter(hasEncoding(_, Encoding.DICTIONARY,
       Encoding.DIRECT_DICTIONARY))
     for (dim <- dimensionsWithDict) {
       breakable {
-        for (i <- 0 until headers.length) {
+        for (i <- headers.indices) {
```
Comment (jackylk) [inline integration/spark/src/main/scala/org/carbondata/spark/util/GlobalDictionaryUtil.scala]: zipWithIndex

File: integration/spark/src/main/scala/org/carbondata/spark/util/GlobalDictionaryUtil.scala [hunk 7]
```diff
@@ -83,41 +84,42 @@ object GlobalDictionaryUtil extends Logging {
...
    */
-  def hasEncoding(dimension: CarbonDimension, encoding: Encoding,
-    excludeEncoding: Encoding): Boolean = {
+  def hasEncoding(dimension: CarbonDimension,
+      encoding: Encoding,
+      excludeEncoding: Encoding): Boolean = {
     if (dimension.isComplex()) {
       var has = false
-      var children = dimension.getListOfChildDimensions
+      val children = dimension.getListOfChildDimensions
```
Comment (jackylk) [inline integration/spark/src/main/scala/org/carbondata/spark/util/GlobalDictionaryUtil.scala]: can we use filter for this branch?

File: integration/spark/src/main/scala/org/carbondata/spark/util/GlobalDictionaryUtil.scala [hunk 8]
```diff
@@ -180,10 +181,9 @@ object GlobalDictionaryUtil extends Logging {
   /**
    * read global dictionary from cache
    */
-  def readGlobalDictionaryFromCache(model: DictionaryLoadModel
-  ): HashMap[String, Dictionary] = {
+  def readGlobalDictionaryFromCache(model: DictionaryLoadModel): HashMap[String, Dictionary] = {
     val dictMap = new HashMap[String, Dictionary]
-    for (i <- 0 until model.primDimensions.length) {
+    for (i <- model.primDimensions.indices) {
```
Comment (jackylk) [inline integration/spark/src/main/scala/org/carbondata/spark/util/GlobalDictionaryUtil.scala]: can use zipWithIndex.filter.map to transform it

File: integration/spark/src/main/scala/org/carbondata/spark/util/GlobalDictionaryUtil.scala [hunk 9]
```diff
@@ -274,30 +273,30 @@ object GlobalDictionaryUtil extends Logging {
...
       }
     }
-    val primDimensions = primDimensionsBuffer.toSeq.map { x => x }.toArray
+    val primDimensions = primDimensionsBuffer.map { x => x }.toArray
     // list dictionary file path
     val dictFilePaths = new Array[String](primDimensions.length)
     val dictFileExists = new Array[Boolean](primDimensions.length)
     val carbonTablePath = CarbonStorePath.getCarbonTablePath(hdfsLocation, table)
-    for (i <- 0 until primDimensions.length) {
+    for (i <- primDimensions.indices) {
```
Comment (jackylk) [inline integration/spark/src/main/scala/org/carbondata/spark/util/GlobalDictionaryUtil.scala]: zipWithIndex for following three for loop?

File: integration/spark/src/main/scala/org/carbondata/spark/util/GlobalDictionaryUtil.scala [hunk 10]
```diff
@@ -417,18 +428,22 @@ object GlobalDictionaryUtil extends Logging {
...
-      else carbonLoadModel.getCsvHeader.split("" + CSVWriter.DEFAULT_SEPARATOR)
+      val headers = if (StringUtils.isEmpty(carbonLoadModel.getCsvHeader)) {
+        df.columns
+      }
+      else {
+        carbonLoadModel.getCsvHeader.split("" + CSVWriter.DEFAULT_SEPARATOR)
+      }
       val (requireDimension, requireColumnNames) = pruneDimensions(dimensions, headers, df.columns)
-      if (requireDimension.size >= 1) {
+      if (requireDimension.length >= 1) {
```
Comment (jackylk) [inline integration/spark/src/main/scala/org/carbondata/spark/util/GlobalDictionaryUtil.scala]: nonEmpty

File: integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonDropCubeRDD.scala (hunk 3 above)
Comment (ravipesala) [inline integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonDropCubeRDD.scala]: it is todo, needs to clear btree from both driver and executor memory after table is dropped.

**Comment by ravipesala** [comment]
@jackylk Rebased and fixed review comments. Please review.

File: integration/spark/src/main/scala/org/carbondata/spark/util/GlobalDictionaryUtil.scala [hunk 11]
```diff
@@ -316,23 +304,27 @@ object GlobalDictionaryUtil extends Logging {
...
    */
   def getCsvRecursivePathsFromCarbonFile(carbonFile: CarbonFile): String = {
-    if (carbonFile.isDirectory()) {
+    if (carbonFile.isDirectory) {
       val files = carbonFile.listFiles()
       val stringbuild = new StringBuilder()
       for (j <- 0 until files.size) {
         val filePath = getCsvRecursivePathsFromCarbonFile(files(j))
-        if (!filePath.isEmpty()) {
+        if (!filePath.isEmpty) {
```
Comment (jackylk) [inline integration/spark/src/main/scala/org/carbondata/spark/util/GlobalDictionaryUtil.scala]: please use nonEmpty like following one

File: integration/spark/src/main/scala/org/carbondata/spark/util/GlobalDictionaryUtil.scala [hunk 12]
```diff
@@ -349,11 +341,11 @@ object GlobalDictionaryUtil extends Logging {
         val fileType = FileFactory.getFileType(filePaths(i))
         val carbonFile = FileFactory.getCarbonFile(filePaths(i), fileType)
         val filePath = getCsvRecursivePathsFromCarbonFile(carbonFile)
-        if (!filePath.isEmpty()) {
+        if (!filePath.isEmpty) {
```
Comment (jackylk) [inline integration/spark/src/main/scala/org/carbondata/spark/util/GlobalDictionaryUtil.scala]: please use nonEmpty like following one

**Comment by jackylk** [comment]
LGTM

**Pull request merged**